modules/net-ilb-l7/README.md

```diff
@@ -0,0 +1,398 @@
+    tests                = null
+    host_rules           = []
+    path_matchers = [
+      {
+        name = "my-example-page"
+        path_rules = [
+          {
+            paths   = ["/my-example-page"]
+            service = "another-group-backend"
+          }
+        ]
+      }
+    ]
+  }
+
+  backend_services_config = {
+    my-backend-svc = {
+      enable_cdn    = false
+      cdn_config    = null
+      backends = [
+        {
+          group   = "my_test_group"
+          options = null
+        }
+      ],
+      health_checks = []
+      log_config = null
+      options = null
+    },
+    my-example-page = {
+      enable_cdn    = false
+      cdn_config    = null
+      backends = [
+        {
+          group   = "my_other_test_group"
+          options = null
+        }
+      ],
+      health_checks = []
+      log_config = null
+      options = null
+    }
+  }
+}
+# tftest modules=1 resources=6
+```
+
+### Reserve a static IP address
+
+Optionally, a static IP address can be reserved:
+
+```hcl
+module "ilb" {
+  source     = "./modules/net-ilb-l7"
+  name       = "ilb-test"
+  project_id = var.project_id
+  region     = "europe-west1"
+
+  static_ip_config = {
+    reserve = true
+    options = null
+  }
+
+  backend_services_config = {
+    my-backend-svc = {
+      enable_cdn    = false
+      cdn_config    = null
+      backends = [
+        {
+          group   = "my_test_group"
+          options = null
+        }
+      ],
+      health_checks = []
+      log_config = null
+      options = null
+    }
+  }
+}
+# tftest modules=1 resources=6
+```
+
+### HTTPS And SSL Certificates
+
+HTTPS is disabled by default but it can be optionally enabled.
+
+When HTTPS is enabled, if the ids specified in the `target_proxy_https_config` variable are not found in the `ssl_certificates_config` map, they are used as is, assuming the ssl certificates already exist:
+
+```hcl
+module "ilb" {
+  source     = "./modules/net-ilb-l7"
+  name       = "ilb-test"
+  project_id = var.project_id
+  region     = "europe-west1"
+
+  https = true
+
+  target_proxy_https_config = {
+    ssl_certificates = [
+      "an-existing-cert"
+    ]
+  }
+
+  backend_services_config = {
+    my-backend-svc = {
+      enable_cdn    = false
+      cdn_config    = null
+      backends = [
+        {
+          group   = "my_test_group"
+          options = null
+        }
+      ]
+      health_checks = []
+      log_config = null
+      options = null
+    }
+  }
+}
+# tftest modules=1 resources=5
+```
+
+Otherwise, unamanged certificates can also be contextually created:
+
+```hcl
+module "ilb" {
+  source     = "./modules/net-ilb-l7"
+  name       = "ilb-test"
+  project_id = var.project_id
+  region     = "europe-west1"
+
+  https = true
+
+  ssl_certificates_config = {
+    my-domain = {
+      domains = [
+        "my-domain.com"
+      ],
+      tls_private_key      = nonsensitive(tls_private_key.self_signed_key.private_key_pem)
```

Comment:
do we really need nonsensitive on the key?

Reply:
I'd leave it as it is. It's consistent with what we already do for the glb and it reflects what's prescribed in the providers readme.